### PR TITLE
Use web3's gas price strategies

### LINKED ISSUE #1473: web3: gas price strategy
Currently we get the gas price from the ethereum node by calling the `eth_gasPrice` function.

Web3 offers different gas price strategies. The one we currently use is available as well as a more sophisticated time based version that makes sure that transactions get mined with a certain probability in a given timeframe. (This is similar to what Eth Gas Station provides)
See https://web3py.readthedocs.io/en/stable/gas_price.html#module-web3.gas_strategies.rpc)

When switch to the time based strategy we should define what our goal reading transaction delays are.

### PR DESCRIPTION
**Pull request by palango**
Use web3's gas price strategies 

Fixes #1473

### CODE REVIEW

**Comment by LefterisJP** [comment]
@palango smart contract tests seem to fail with tester

**Comment by palango** [comment]
@LefterisJP Yes, but I didn't want to spend more time on this right now.

File: raiden/network/rpc/client.py [hunk 1]
```diff
@@ -217,12 +224,10 @@ def gaslimit(self, location='latest') -> int:
         gas_limit = self.web3.eth.getBlock(location)['gasLimit']
         return gas_limit * 8 // 10
 
-    @cachedmethod(attrgetter('_gasprice_cache'))
```
Comment (andrevmatos) [inline raiden/network/rpc/client.py]: You should also remove `self._gasprice_cache` then

File: raiden/network/rpc/client.py [hunk 2]
```diff
@@ -217,12 +224,10 @@ def gaslimit(self, location='latest') -> int:
         gas_limit = self.web3.eth.getBlock(location)['gasLimit']
         return gas_limit * 8 // 10
```
Comment (andrevmatos) [inline raiden/network/rpc/client.py]: Any chance of also including better gas limit calculations on this PR? =)

File: raiden/network/rpc/client.py (hunk 1 above)
Comment (palango) [reply raiden/network/rpc/client.py]: Good catch

File: raiden/network/rpc/client.py (hunk 2 above)
Comment (palango) [reply raiden/network/rpc/client.py]: No, not that I know of. What do you mean exactly?

Comment (palango) [reply raiden/network/rpc/client.py]: Mh, seems like `gaslimit` is only used in the assumption tests anyways. we can probably remove that all together.

**Pull request merged**